crates/task-impls/src/quorum_proposal.rs

```diff
@@ -80,8 +71,160 @@ struct ProposalDependencyHandle<TYPES: NodeType> {
     sender: Sender<Arc<HotShotEvent<TYPES>>>,
 
     /// Reference to consensus. The replica will require a write lock on this.
-    #[allow(dead_code)]
     consensus: Arc<RwLock<Consensus<TYPES>>>,
+
+    /// Output events to application
+    #[allow(dead_code)]
+    output_event_stream: async_broadcast::Sender<Event<TYPES>>,
+
+    /// Membership for Timeout votes/certs
+    #[allow(dead_code)]
+    timeout_membership: Arc<TYPES::Membership>,
+
+    /// Membership for Quorum Certs/votes
+    quorum_membership: Arc<TYPES::Membership>,
+
+    /// Our public key
+    public_key: TYPES::SignatureKey,
+
+    /// Our Private Key
+    private_key: <TYPES::SignatureKey as SignatureKey>::PrivateKey,
+
+    /// View timeout from config.
+    #[allow(dead_code)]
+    timeout: u64,
+
+    /// Round start delay from config, in milliseconds.
+    round_start_delay: u64,
+
+    /// The node's id
+    #[allow(dead_code)]
+    id: u64,
+}
+
+impl<TYPES: NodeType> ProposalDependencyHandle<TYPES> {
+    /// Sends a proposal if possible from the high qc we have
+    #[allow(clippy::too_many_lines)]
+    pub async fn publish_proposal_if_able(
+        &self,
+        view: TYPES::Time,
+        event_stream: &Sender<Arc<HotShotEvent<TYPES>>>,
+        commit_and_metadata: CommitmentAndMetadata<TYPES::BlockPayload>,
+    ) -> bool {
+        if self.quorum_membership.get_leader(view) != self.public_key {
+            // This is expected for view 1, so skipping the logging.
+            if view != TYPES::Time::new(1) {
+                error!(
+                    "Somehow we formed a QC but are not the leader for the next view {:?}",
+                    view
+                );
+            }
+            return false;
+        }
+
+        let consensus = self.consensus.read().await;
+        let parent_view_number = &consensus.high_qc.get_view_number();
+        let mut reached_decided = false;
+
+        let Some(parent_view) = consensus.validated_state_map.get(parent_view_number) else {
+            // This should have been added by the replica?
+            error!("Couldn't find parent view in state map, waiting for replica to see proposal\n parent view number: {}", **parent_view_number);
+            return false;
+        };
```

Comment:
Does this mean the quorum proposal task will not propose on `QCFormed` if it missed the parent? 

I believe this is different from what consensus currently does, where we currently can propose off a formed QC even if we never saw the parent.

(I'm not sure exactly what we're planning to have handled by which task, just pointing out that I think this is different)